crates/router/src/routes/routing.rs

```diff
@@ -688,6 +690,167 @@ pub async fn retrieve_surcharge_decision_manager_config(
     .await
 }
 
+#[cfg(all(feature = "olap", feature = "v1"))]
+pub async fn retrieve_linked_surcharge_config(
+    state: web::Data<AppState>,
+    req: HttpRequest,
+    query: web::Query<routing_types::SurchargeRetrieveLinkQuery>,
+) -> impl Responder {
+    let flow = Flow::DecisionManagerRetrieveConfig;
+    let query = query.into_inner();
+    Box::pin(oss_api::server_wrap(
+        flow,
+        state,
+        &req,
+        query.clone(),
+        |state, auth: auth::AuthenticationData, query, _| {
+            surcharge_decision_config::retrieve_surcharge_config(
+                state,
+                auth.merchant_account,
+                auth.key_store,
+                query,
+            )
+        },
+        #[cfg(not(feature = "release"))]
+        auth::auth_type(
+            &auth::HeaderAuth(auth::ApiKeyAuth),
+            &auth::JWTAuthProfileFromRoute {
+                profile_id: query.profile_id,
+                required_permission: Permission::ProfileRoutingRead,
+            },
+            req.headers(),
+        ),
+        #[cfg(feature = "release")]
+        &auth::JWTAuthProfileFromRoute {
+            profile_id: query.profile_id,
+            required_permission: Permission::ProfileRoutingRead,
+        },
+        api_locking::LockAction::NotApplicable,
+    ))
+    .await
+}
+
+#[cfg(all(feature = "olap", feature = "v1"))]
+pub async fn list_surcharge_decision_manager_configs(
+    state: web::Data<AppState>,
+    req: HttpRequest,
+    query: web::Query<routing_types::SurchargeRetrieveQuery>,
+    algorithm_type: enums::AlgorithmType,
+) -> impl Responder {
+    let flow = Flow::DecisionManagerRetrieveConfig;
+    Box::pin(oss_api::server_wrap(
+        flow,
+        state,
+        &req,
+        query.into_inner(),
+        |state, auth: auth::AuthenticationData, request, _| {
+            surcharge_decision_config::list_surcharge_decision_configs(
+                state,
+                auth.profile_id,
+                request.limit.unwrap_or_default(),
+                request.offset.unwrap_or_default(),
+                algorithm_type,
+            )
+        },
+        #[cfg(not(feature = "release"))]
+        auth::auth_type(
+            &auth::HeaderAuth(auth::ApiKeyAuth),
+            &auth::JWTAuth {
+                permission: Permission::ProfileSurchargeDecisionManagerRead,
+            },
+            req.headers(),
+        ),
+        #[cfg(feature = "release")]
+        &auth::JWTAuth {
+            permission: Permission::ProfileSurchargeDecisionManagerWrite,
+        },
```

Comment:
Why are we using the `ProfileSurchargeDecisionManagerWrite` permission when the `release` feature is enabled but `ProfileSurchargeDecisionManagerRead` when it is disabled? Shouldn't we use `ProfileSurchargeDecisionManagerRead` in both cases?